sdk/core/wasm/src/file_metadata.rs

```diff
@@ -0,0 +1,58 @@
+use wasm_bindgen::prelude::*;
+
+use parity_scale_codec::Encode;
+use shp_constants::{FILE_CHUNK_SIZE, FILE_SIZE_TO_CHALLENGES, H_LENGTH};
+use shp_file_metadata::{FileMetadata as RustFileMetadata, Fingerprint};
+use sp_core::hashing::blake2_256;
+
+// ────────────────────────────────────────────────────────────────────────────
+// WASM‐exposed wrapper for FileMetadata
+// ────────────────────────────────────────────────────────────────────────────
+#[wasm_bindgen]
+pub struct FileMetadata {
+    inner: RustFileMetadata<{ H_LENGTH }, { FILE_CHUNK_SIZE }, { FILE_SIZE_TO_CHALLENGES }>,
+}
+
+#[wasm_bindgen]
+impl FileMetadata {
+    /// Constructs a new `FileMetadata`.
+    /// * `owner`, `bucket_id`, `fingerprint` – 32-byte arrays (passed as slices)
+    /// * `location` – arbitrary byte string (file path)
+    /// * `size` – file size in bytes
+    #[wasm_bindgen(constructor)]
+    pub fn new(
+        owner: &[u8],
+        bucket_id: &[u8],
+        location: &[u8],
+        size: u64,
+        fingerprint: &[u8],
+    ) -> Result<FileMetadata, JsValue> {
+        // Basic validation to return early, preventing panics deeper inside.
+        if owner.len() != H_LENGTH
+            || bucket_id.len() != H_LENGTH
+            || fingerprint.len() != H_LENGTH
+            || location.is_empty()
+            || size == 0
+        {
+            return Err(JsValue::from_str("Invalid FileMetadata parameters"));
```

Comment:
I would be more specific here to return an error message based on what invalid parameters were passed. 

I imagine being a user of the SDK and not understanding what I might have done wrong.